lib/acts_as_list/active_record/acts/list.rb

```diff
@@ -84,7 +84,7 @@ def acts_as_list_top
             end
 
             def acts_as_list_class
-              self.class
+              ::#{self.name}
```

Comment:
Ah, yeah. So there are two things going on there.

One is that we're inside a `class_eval` metaprogramming block, so any string interpolation with `#{}` gets applied directly to the ruby code being constructed. So if the value of `self.name` is `"TheBaseClass"` when the `acts_as_list` macro is invoked, that line compiles to `::TheBaseClass`.

Second is `::`, which is the scope resolution operator. This is there to make sure that we are definitely getting the "root-level" `TheBaseClass`, instead of unintentionally getting something like `SomeOtherModuleThatAlsoContains::TheBaseClass`.

Does this help?